libafl/src/executors/mod.rs

```diff
@@ -71,45 +66,15 @@ pub enum ExitKind {
     // Custom(Box<dyn SerdeAny>),
 }
 
-/// How one of the diffing executions finished.
-#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq)]
-#[cfg_attr(
-    any(not(feature = "serdeany_autoreg"), miri),
-    allow(clippy::unsafe_derive_deserialize)
-)] // for SerdeAny
-pub enum DiffExitKind {
-    /// The run exited normally.
-    Ok,
-    /// The run resulted in a target crash.
-    Crash,
-    /// The run hit an out of memory error.
-    Oom,
-    /// The run timed out
-    Timeout,
-    /// One of the executors itelf repots a differential, we can't go into further details.
-    Diff,
-    // The run resulted in a custom `ExitKind`.
-    // Custom(Box<dyn SerdeAny>),
-}
+/// The type of the crash observed in one of the differential executors
+pub type DiffExitKind = Box<ExitKind>;
```

Comment:
Why heap allocate this thing?

Reply:
It allows us to recursively enter the Diff types, instead of the old behaviour which drops that information.

Reply:
Is it worth the amount of extra allocations? Any way to reuse an existing buf to store that information?

Reply:
No, because it can be recursive. However, it shouldn't matter, because this will only affect users using diff executors (which will have pretty severe overhead anyways). Otherwise, this will get reduced to a copy.

Reply:
I mean it could make sense to meaningful reduce one side of the recursion then, since intermediate results may not be relevant, but also it's probably a very nieche use case

Reply:
One thing though: it could be `Copy` making this dude potentially faster in all cases(?)

Reply:
If we don't assume super smart compilers

Reply:
I think we can safely assume smart compilers :grin: